In [12]:
import numpy as np
from math import factorial

Если проводится большое количество испытаний $n$ и при этом вероятность $p$ появления события $A$ в отдельном испытании мала, применяют формулу Пуассона для вычисления вероятности того, что событие произойдет $m$ раз в $n$ испытаниях:

$$P_m \approx \frac{\lambda^m}{m!}e^{-\lambda}$$

λ = p*n 

In [5]:
def puasson(n, m, p):
    lambda_ = n * p
    return (lambda_**m / np.math.factorial(m)) * np.exp(-lambda_)

In [13]:
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

### 1. Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.

Применим формулу Пуассона:

In [6]:
n = 100
m = 85
p = 0.8

puasson(n, m, p)

0.03709261434369179

**Ответ**: Вероятность, что стрелок попадет в цель ровно 85 раз: 0.037 или 3,7%

### 2. Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. Какова вероятность, что ни одна из них не перегорит в первый день? Какова вероятность, что перегорят ровно две?

In [8]:
n = 5000
m = 0
p = 0.0004

puasson(n, m, p)

0.1353352832366127

In [9]:
n = 5000
m = 2
p = 0.0004

puasson(n, m, p)

0.2706705664732254

**Ответ**: Вероятность, что ни одна лампочка не перегорит: 0.135 или 13,5%.
        Вероятность, что перегорят ровно 2 лампочки: 0.2706 или 27,1%

### 3. Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

In [49]:
combinations(144, 70)*(0.5**70)*((1-0.5)**(144-70))

0.06281178035144776

**Ответ**: Вероятность, что орел выпадет ровно 70 раз: 0.0628 или 6,3%.

### 4. В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. Какова вероятность того, что все мячи белые? Какова вероятность того, что ровно два мяча белые? Какова вероятность того, что хотя бы один мяч белый?

Все мячи белые:

In [50]:
7/10*6/9*9/11*8/10

0.3054545454545454

**Ответ**: Вероятность, что все мячи белые: 0.3054 или 30,54%.

Оба меча будут белые, если 2 белых вытащить из первого ящика и 0 из второго ИЛИ 1 белый из первого и 1 белый из второго ИЛИ 0 белых из первого и 2 белых из второго.


In [40]:
# P где 2 белых из первого ящика и 2 черных из второго ящика:
first_case = ((7/10)*(6/9))*((2/11)*(1/10))
first_case

0.008484848484848486

In [51]:
# P где 1 белый из первого ящика и 1 белый из второго + умножение на 4 так как у нас вариантов сочетаний ч и б 4 разных:
second_case = ((7/10)*(3/9))*((9/11)*(2/10))*4
second_case

0.15272727272727274

In [42]:
# P где 0 белых из первого ящика и 2 белых из второго:
third_case = ((3/10)*(2/9))*((9/11)*(8/10))
third_case

0.04363636363636364

Таким образом вероятности посчитаны для каждого случая, объединим их:

In [52]:
P = first_case + second_case + third_case
P

0.20484848484848486

**Ответ**: Вероятность, что ровно два мяча белые: 0.2048 или 20,48%.

Хотя бы 1 мяч белый будет если из первой корзины вытащить 1 белый/1 черный и из второй оба черные ИЛИ наоборот:

Решим задачу наоборот от варианта что все черные:

In [54]:
P = 1 - ((3/10)*(2/9)*(2/11)*(1/10))
P

0.9987878787878788

**Ответ**: Вероятность, что хотя бы один белый: 0.9987 или 99,87%.